src/jpegxl.imageio/jxlinput.cpp

```diff
@@ -359,6 +381,12 @@ JxlInput::close()
     if (ioproxy_opened()) {
         close_file();
     }
+
+    if (m_buffer) {
+        m_buffer.reset();
+        m_buffer = nullptr;
+    }
+
```

Comment:
```suggestion
    m_buffer.reset();
```

Don't need the assignment or even the check.